### ISSUE TITLE
This changes seems to break previous cors configuration and I couldn't find documentation about how to upgrade to the new changes.

### ISSUE THREAD

**Issue by Pablo-Aldana** [strapi/strapi#20544]
This changes seems to break previous cors configuration and I couldn't find documentation about how to upgrade to the new changes.

Right now I have my CORS like this:

```
{
    name: 'strapi::cors',
    config: {
      //enabled: true,
      headers: '*',
      origin: env('CORS'),
    },
  },
  ```

And in the env(CORS) like this but with PROD values:
CORS=http://localhost:1338, http://localhost:8000, http://192.168.68.114:8000, http://192.168.68.114:3000, http://localhost:3000, http://192.168.68.114:1338

As soon as I update to this the strapi version 4.24.0 (https://github.com/strapi/strapi/releases/tag/v4.24.0)or higher, the web stop working due to cors issues, even if the URL is in the cors.

![Screenshot_2024-06-13 23 40 57_studUG](https://github.com/strapi/strapi/assets/9969839/6b1b817f-fa54-45e5-a4ec-66d7037bdb82)

Can anyone point me to the documentation or  update guide for this change?
I can create a new issue if needed

Many thanks

_Originally posted by @Pablo-Aldana in https://github.com/strapi/strapi/issues/20044#issuecomment-2166888703_

**Comment by alexandrebodin** [comment]
Coping the answer from the PR comment here :)

With the cors fix we had to do for security we removed the string list support as we don't know if it makes sense to process it this way or not. to fix this for your usecase you should split the string yourself

env('CORS').split(', ')

**Comment by alexandrebodin** [comment]
If we can support a string list again we will add it back when the issue is triaged & prioritize 🙏

**Comment by Pablo-Aldana** [comment]
Hi @alexandrebodin  thanks a lot for your supper I will definitely try this. 

Not sure what would be the best approach, but I think the migration guide and the documents should get updated as from my side we got a massive issue in production because of this even tho every time we upgrade we check the migration and upgrade guide.

**Comment by alexandrebodin** [comment]
cc @pwizla were do you think we could add some guidance after that security release ?

**Comment by pwizla** [comment]
Hi @alexandrebodin, the documentation for `cors` is [here](https://docs.strapi.io/dev-docs/configurations/middlewares#cors). I think that if you plan to support string lists again in the future, maybe a [migration guide](https://docs.strapi.io/dev-docs/migration-guides#v4-migration-guides) is not necessary. If you feel like it is, I can create a template and I'd love if you could fill in the blanks as I'm not sure what needs to be documented here 🤔

**Comment by alexandrebodin** [comment]
Yeah I didn't recall documenting that we supported string array in the 1st place. I'll see what's best if we can support it again it will be better

**Comment by derrickmehaffy** [comment]
Yeah I'm like 99% sure we never supported comma separated values, only arrays.

If you do `CORS=[http://localhost:1338/, http://localhost:8000/, http://192.168.68.114:8000/, http://192.168.68.114:3000/, http://localhost:3000/, http://192.168.68.114:1338/]` and make sure you cast the env var to an array like `origin: env.array("CORS")` it should work fine

**Comment by alexandrebodin** [comment]
Hi @Pablo-Aldana fyi  I opened a fix PR here https://github.com/strapi/strapi/pull/20658 if you want to try and validate it works for you.you can try this version `0.0.0-experimental.d9f9f725e07a329ca272ecedef7bd85fa74dd661`
